packages/react/src/exchanges/hooks/useExchangeFilters.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  createExchangeFilter as createExchangeFilterAction,
+  getExchangeFilterError,
+  getExchangeFilters,
+  isExchangeFilterLoading,
+  resetExchangeFilters,
+  type StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction.js';
+import type {
+  Config,
+  ExchangeFilterItem,
+  OrderItem,
+  PostExchangeFilterData,
+} from '@farfetch/blackout-client';
+
+type PostExchangeFilterDataHook = PostExchangeFilterData & {
+  exchangeFilterItems: Array<ExchangeFilterItemWithOptionalUuid>;
+};
+
+type ExchangeFilterItemWithOptionalUuid = Omit<
+  ExchangeFilterItem,
+  'orderItemUuid'
+> & {
+  orderItemUuid?: OrderItem['shippingOrderLineId'];
+};
+
+/**
+ * Provides facilities to create an exchange filter.
+ */
+function useExchangeFilters(orderItemUuid?: OrderItem['shippingOrderLineId']) {
+  const orderItemUuidHookParameter = orderItemUuid;
+  const isCreating = useSelector((state: StoreState) =>
+    orderItemUuidHookParameter
+      ? isExchangeFilterLoading(state, orderItemUuidHookParameter)
+      : false,
+  );
+  const error = useSelector((state: StoreState) =>
+    orderItemUuidHookParameter
+      ? getExchangeFilterError(state, orderItemUuidHookParameter)
+      : null,
+  );
+  const exchangeFilters = useSelector(getExchangeFilters);
+  const createExchangeFilter = useAction(createExchangeFilterAction);
+  const reset = useAction(resetExchangeFilters);
+
+  /**
+   * Creates an exchange filter. This method will save the exchange filter by orderItemUuid without
+   * replacing existing filters. Will throw if no exchangeFilterData is passed
```

Comment:
```suggestion
   * replacing existing filters. Will throw in the following conditions:
     - exchangeFilterData parameter is not passed;
     - or no usable `orderItemUuid` is found. If it is not passed in `exchangeFilterData`, the one passed on the hook call will be used.
```

Reply:
I would also add this: 

`Please notice that if you pass a value for the orderItemUuid property inside exchangeFilterData parameter, the values for isLoading, error and exchangeFilters properties returned by the hook will not be based on it as the hook bases those values on the value of the orderItemUuid that was passed to the hook. If you need to call the create action with a different value for the orderItemUuid property that was passed to the hook, do not rely on the values returned by hook for the isLoading, error and exchangeFilters properties.`

Reply:
Done